src/ripple/app/misc/Validations.cpp

```diff
@@ -429,25 +429,34 @@ class ValidationsImp : public Validations
         bool anyNew = false;
 
         JLOG (j_.info()) << "Flushing validations";
-        ScopedLockType sl (mLock);
-        for (auto& it: mCurrentValidations)
         {
-            if (it.second)
-                mStaleValidations.push_back (it.second);
-
-            anyNew = true;
-        }
-        mCurrentValidations.clear ();
+            ScopedLockType sl (mLock);
+            for (auto& it: mCurrentValidations)
+            {
+                if (it.second)
+                {
+                    mStaleValidations.push_back (it.second);
+                    anyNew = true;
+                }
+            }
+            mCurrentValidations.clear ();
 
-        if (anyNew)
-            condWrite ();
+            // If there isn't a write in progress already, then write to the
+            // database synchronously.
+            if (anyNew && !mWriting)
+            {
+                mWriting = true;
+                doWrite (sl);
+            }
 
-        while (mWriting)
-        {
-            ScopedUnlockType sul (mLock);
-            std::this_thread::sleep_for (std::chrono::milliseconds (100));
+            // Handle the case where flush() is called while a queuedWrite
+            // is already in progress.
```

Comment:
I am not sure I understand the comment. It seems that if queudWrite was in progress it would hold the lock and we could never arrive here at the same time.